sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -162,22 +162,36 @@ case class HiveTableScanExec(
     }
   }
 
-  // exposed for tests
-  @transient lazy val rawPartitions = {
-    val prunedPartitions =
-      if (sparkSession.sessionState.conf.metastorePartitionPruning &&
-          partitionPruningPred.nonEmpty) {
-        // Retrieve the original attributes based on expression ID so that capitalization matches.
-        val normalizedFilters = partitionPruningPred.map(_.transform {
-          case a: AttributeReference => originalAttributes(a)
-        })
-        relation.prunedPartitions.getOrElse(
-          sparkSession.sessionState.catalog
-            .listPartitionsByFilter(relation.tableMeta.identifier, normalizedFilters))
+  @transient lazy val prunedPartitions: Seq[HivePartition] = {
+    if (relation.prunedPartitions.nonEmpty &&
```

Comment:
I think the logic should be
```
if (relation.prunedPartitions.nonEmpty) {
  if (no subquery predicate) {
    relation.prunedPartitions
  } else {
    prunePartitions(relation.prunedPartitions)
  }
} else {
  val rawPartitions = prunedPartitionsViaHiveMetaStore()
  prunePartitions(relation.prunedPartitions)
}
```

Reply:
> else {
>   val rawPartitions = prunedPartitionsViaHiveMetaStore()
>   prunePartitions(relation.prunedPartitions)
> }

Since prunedPartitionsViaHiveMetaStore will return None if the HIVE_METASTORE_PARTITION_PRUNING is disabled, so we need to check the result of prunedPartitionsViaHiveMetaStore here.

Reply:
@cloud-fan Refined per my understanding, please correct me if i am wrong. thanks.